lib/Server.js

```diff
@@ -20,6 +20,13 @@ class Server {
     // TODO: remove this after plugin support is published
     if (options.hooks) {
       [options, compiler] = [compiler, options];
+
+      this.compiler = compiler;
+      this.logger = this.compiler.getInfrastructureLogger("webpack-dev-server");
```

Comment:
don't do it, you should use https://nodejs.org/api/util.html#util_util_deprecate_fn_msg_code

Reply:
Not sure how to use `util.deprecate` on the constructor, I used `process.emitWarning` instead - 

<img width="951" alt="Screenshot 2021-07-24 at 9 42 18 AM" src="https://user-images.githubusercontent.com/46647141/126856977-6b0b1cfa-0df8-4e04-b9a9-72b33eaf3f12.png">

Reply:
@snitin315 You can pass an empty function to `util.deprecate()` and call it.

Reply:
@ylemkimon thanks 👍🏻